pixie/stdlib.pxi

```diff
@@ -1304,9 +1312,13 @@ and implements IAssociative, ILookup and IObject."
                                           fields)))
                         `(-hash [self]
                                 (hash [~@field-syms]))]
-        deftype-decl `(deftype ~nm ~fields ~@default-bodies ~@body)]
+        deftype-decl `(deftype ~nm ~(conj fields '__meta) ~@default-bodies ~@body)
+        ctor `(defn ~ctor-name ~field-syms
```

Comment:
Just to expand on that, I didn't do it in deftype, because types should be able to choose not to implement IMeta (like in clojure), and be more lightweight this way (it's one less field after all).